File: backend/src/routes/usage.rs
use axum::{
    extract::State,
    routing::get,
    Json, Router,
};

use crate::auth::OptionalAuthUser;
use crate::db::{self, DbPool, DailyUsageStatus};
use crate::models::ApiResponse;

pub fn routes(db_pool: DbPool) -> Router {
    Router::new()
        .route("/status", get(get_usage_status))
        .with_state(db_pool)
}

/// Get current usage status for the user
async fn get_usage_status(
    State(pool): State<DbPool>,
    auth: OptionalAuthUser,
) -> Json<ApiResponse<DailyUsageStatus>> {
    let user_id = auth.user_id_or_default();

    match db::get_daily_usage(&pool, user_id).await {
        Ok(status) => Json(ApiResponse::success(status)),
        Err(e) => Json(ApiResponse::error(format!("Failed to get usage status: {}", e))),
    }
}

File: worker/src/routes/usage.ts
import { Hono } from 'hono'
import type { Env } from '../env'
import { requireAuth } from '../middleware/auth'
import * as db from '../db'

export function usageRoutes() {
  const app = new Hono<{ Bindings: Env }>()

  // ─── Get usage status ───
  app.get('/status', async (c) => {
    const auth = await requireAuth(c)
    if (auth instanceof Response) return auth

    try {
      const usage = await db.getDailyUsage(c.env.DB, auth.user_id)
      return c.json({ success: true, data: usage })
    } catch (e) {
      return c.json({ success: false, error: `Failed to get usage: ${e}` })
    }
  })

  return app
}
